proposals/1533-host-specific-tasklist.md

```diff
@@ -0,0 +1,241 @@
+
+The following sections will first go through the APIs exposed to users, provide some explanation on how to use them and some sample code. It should be very easy for any developer with previous cadence experience to understand those APIs and start using them. After that I will explain the model behind those APIs and how they are implemented.
+
+### API Changes
+
+#### New Go Client API
+
+There will be four new APIs available when writing a workflow:
+
+```go
+sessionCtx, err := workflow.**CreateSession**(ctx)
+```
+
+User call this API to create a session and use the returned sessionCtx to execute activities. All activities executed within returned sessionCtx are considered to be part of the session and will be executed on the same worker. It’s possible that this call will fail due to various reasons: for example, no worker is available at this time.
+
+The session will be created on one of the workers that poll from the tasklist specified in the context passed into the CreateSession API. As a result, users needs to make sure they are using the correct task list.
+
+When executing an activity within a session, user may get two types of errors. The first one is those returned from user activities. Session will not be marked as failed in this case, so user can return whatever error they want and apply their business logic as necessary. If they want to end a session due to the error returned from the activity, use the CompleteSession() API below. The second type of error is a special error which says the session has failed due to worker failure and the session is marked as failed in the background. In this case, no activities can be executed using this context. It’s the user’s choice on how to handle the failure; for example, they can create a new session to retry or end the workflow with an error.
+
+```go
+err := workflow.**CompleteSession**(sessionCtx)
+```
+
+This API is used to complete a session. It will return an error if the context passed in does not contain a session with state "Open" or it fails to complete the session.
+
+```go
+sessionInfo, err := workflow.**GetSessionInfo**(sessionCtx)
+```
+
+This API returns session information stored in the context. It returns an error if the context passed in doesn’t contain any session information or the session has already failed. For now, the only information returned is sessionID, which is a unique identifier for a session.
+
+```go
+sessionCtx, err := workflow.**RecreateSession**(ctx, sessionInfo)
+```
+
+This API takes session information and recreate a session. All the activities in the recreated session will be executed by the specified worker. The primary use case will be continueAsNew. Before returning the continueAsNew error, user should get the sessionInfo from sessionCtx and complete the session. Then in the next run, the session can be recreated and activities in this new run can continue to be executed on the same activity worker as previous runs.
```

Comment:
recreate -> recreates
user should get the sessionInfo -> the user should get the sessionInfo